.github/workflows/test-runner-integration.yml

```diff
@@ -0,0 +1,56 @@
+# This workflow will build a Java project with Gradle
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-gradle
+
+name: Run Test Runner Tests
+
+on:
+  push:
+    paths-ignore:
+      - 'README.md'
+      - 'local-dev/**'
+
+jobs:
+  test-runner-perf:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up AdoptOpenJDK 11
+        uses: joschi/setup-jdk@v2
+        with:
+          java-version: 11
+      - name: Get Vault token
+        id: vault-token-step
+        env:
+          VAULT_ADDR: https://clotho.broadinstitute.org:8200
+        run: |
+          VAULT_TOKEN=$(docker run --rm --cap-add IPC_LOCK \
+            -e "VAULT_ADDR=${VAULT_ADDR}" \
+            vault:1.1.0 \
+            vault write -field token \
+              auth/approle/login role_id=${{ secrets.VAULT_APPROLE_ROLE_ID }} \
+              secret_id=${{ secrets.VAULT_APPROLE_SECRET_ID }})
+          echo ::set-output name=vault-token::$VAULT_TOKEN
+          echo ::add-mask::$VAULT_TOKEN
+      - name: "Configure Test Runner"
+        run: |
+          cd ${GITHUB_WORKSPACE}/${workingDir}/workspace-manager-clienttests
+          chmod +x render-config.sh
+          ./render-config.sh ${{ steps.vault-token-step.outputs.vault-token }}
+          chmod +r rendered
+          ls -l rendered
+      - name: Cache Gradle packages
+        uses: actions/cache@v2
+        with:
+          path: |
+            .gradle/caches
+            .gradle/wrapper
+          key: v1-${{ runner.os }}-gradle-${{ hashfiles('**/gradle-wrapper.properties') }}-${{ hashFiles('**/*.gradle') }}
+          restore-keys: v1-${{ runner.os }}-gradle-${{ hashfiles('**/gradle-wrapper.properties') }}
+      - name: Run GetWorkspace tests using test runner library
+        run: |
+          cd ${GITHUB_WORKSPACE}/${workingDir}/workspace-manager-client
+          ../gradlew clean
+          ../gradlew build
+          cd ${GITHUB_WORKSPACE}/${workingDir}/workspace-manager-clienttests
+          ./gradlew runTest --args="configs/BasicAuthenticated.json /tmp/TestRunnerResults"
```

Comment:
this will run an arbitrary number of concurrent perf tests against the dev workspace manager (and dev sam, etc). Since this action is triggered on every `push`, we can have many tests running at once - as many as we have commits to open PRs plus merges to dev, etc. Is that what we really want to do? It feels like that could cause lots of variability in test results.

Reply:
I don't think there's any intention to be running perf tests (yet) on every PR push. I'd expect this execution mode to be scoped only to integration-style tests.

There's a separate discussion, picking up some steam, around how exactly we handle issues like this (contention for integration environments from test runners) while we're still waiting on preview environments. See [Service Test Topologies](https://docs.google.com/document/d/13jwti5yTZAKpV2OIM6Rr3e79CdiFQF--8VnuFvEFSXM/edit?resourcekey=0-ByHOWp_X5-bqKSNlqL4CPQ#heading=h.htiffe9pigic) which I'm currently touching up in prep to share more widely and start moving forward on. Might be a good topic for the WSM sync tomorrow.

Reply:
As is, this PR has
```
on:
  push:
    ...
```
so it *will* run perf tests on every push. If that's not the intention, this PR will have to change its trigger. I am considering the `configs/BasicAuthenticated.json` as a perf test; are you suggesting it's not?

Reply:
Right – good point. My working assumption here is that the suite of tests to be run during PR review won't include performance tests, so if that's not reflected here, we should fix it. One thing @ichengchang that might help would be to be very explicit in the config names, whether a given suite of tests is Integration vs. Performance. Perhaps a common prefix, or folder, e.g.

configs/IntegrationBasic.json, or
configs/integration/BasicAuthenticated.json

(and eventually...)

configs/PerfSmokeTests.json
configs/PerfFullSuite.json, or
configs/perf/SmokeTests.json

Reply:
I would prefer the folder structure.

Reply:
Created separate subfolders (`integration`, `perf`) as suggested.